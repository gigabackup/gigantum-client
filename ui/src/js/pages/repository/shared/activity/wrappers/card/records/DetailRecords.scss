@import '~Styles/_imports.scss';

.DetailsRecords {
  overflow: visible;

  &:before {
    content: "";
    display: block;
    position: absolute;
    z-index: 3;
    border-right: 1px dashed $grey;
    height: 10px;
    width: 1px;
    top: 39px;
    left: 25px;
  }

  &__list {
    margin: 0;
    position: relative;
    height: auto;
    max-height: 400px;
    overflow: hidden !important;

    &--long {
      max-height: 10000px !important;
      overflow: visible;
    }
  }


  &__link {
    text-align: right;
    margin: .25em 20px;
    color: $primaryText;
    position: absolute;
    bottom: 0px;
    right: 0px;


    &:hover {
      color: $secondary;
      cursor: pointer;
    }

    &-clicked {
      text-align: right;
      margin-right: 20px;
      color: $primaryText;
    }

    &-clicked:hover {
      color: $secondary;
      cursor: pointer;
    }
  }

  &__fadeout {
    position: absolute;
    bottom: 1px;
    width: 100%;
    height: 3em;
    background: -webkit-linear-gradient(rgba(255,255,255,0) 0%, #fff 100%);
  }

  &__loader {
    margin: 10px 10px 10px 66px;
    width: calc(100% - 66px);
    height: 2px;
    background-color: $primaryText;
    opacity: 0.1;
    animation: fluxuateLoadCards 3s infinite;
  }
}

.LargeIcon {
  background-size: 38px;
}

.SmallIcon {
  background-size: 22px !important;
}


.ActivityDetail__details.note .Record__container:first-child:before {
  display: none;
}
