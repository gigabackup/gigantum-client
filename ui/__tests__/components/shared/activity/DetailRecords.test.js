// vendor
import React from 'react';
import renderer from 'react-test-renderer';
import relayTestingUtils from '@gigantum/relay-testing-utils';
// components
import DetailRecords from 'Pages/repository/shared/activity/wrappers/card/records/DetailRecords';
// data
import json from './__relaydata__/DetailRecords.json';


let fixtures = {
  keys: [],
  sectionType: 'labbook',
  detailRecords: [['text/hml', 'this is example Text']],
};

global.data['DetailRecordsQuery'] = json.data

describe('DetailRecords', () => {
  it('Renders a snapshot', () => {
    const wrapper = renderer.create(relayTestingUtils.relayWrap(
       <DetailRecords {...fixtures} />, {}, json.data.labbook)
    );

    const tree = wrapper.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
