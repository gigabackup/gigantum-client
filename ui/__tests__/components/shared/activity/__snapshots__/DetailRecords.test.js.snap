// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailRecords Renders a snapshot 1`] = `
<div
  className="DetailsRecords"
  data-selenium-id="DetailRecordsx"
>
  <div
    className="DetailsRecords__list"
  >
    <ul
      className="Record__container"
      data-selenium-id="Recordx"
    >
      <div
        className="Record__action Record__action--undefined"
      />
    </ul>
  </div>
</div>
`;
